### PR TITLE
Add Hatch (HATCH)

### PR DESCRIPTION
**Pull request by hatchpay**

### CODE REVIEW

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/hatch/HatchExtension.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                String preferredFiatCurrency = FiatCurrency.USD.getCode();
+                String apiKey = null;
+                if (st.hasMoreTokens()) {
+                    preferredFiatCurrency = st.nextToken().toUpperCase();
+                }
+                if (st.hasMoreTokens()) {
+                    apiKey = st.nextToken();
+                }
+                return new CoinmarketcapRateSource(apiKey, preferredFiatCurrency);
+            } else if ("coingecko".equalsIgnoreCase(rsType)) {
```
Comment (generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/hatch/HatchExtension.java]: Please remove. It is provided by different extension.

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/hatch/HatchExtension.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            String preferredFiatCurrency = st.hasMoreTokens() ? st.nextToken().toUpperCase() : FiatCurrency.USD.getCode();
+            return new CoinGeckoRateSource(preferredFiatCurrency);
+        } else if ("coinpaprika".equalsIgnoreCase(rsType)) {
```
Comment (generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/hatch/HatchExtension.java]: Please remove. It is provided by different extension.

**Comment by generalbytes** [comment]
Please resolve previous comments in code.

**Comment by hatchpay** [comment]
> Please resolve previous comments in code.

Done

**Pull request merged**